tests/e2e/certrotation/certrotation_test.go

```diff
@@ -0,0 +1,112 @@
+package secretsencryption
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/k3s-io/k3s/tests/e2e"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// Valid nodeOS: generic/ubuntu2004, opensuse/Leap-15.3.x86_64
+var nodeOS = flag.String("nodeOS", "generic/ubuntu2204", "VM operating system")
+var serverCount = flag.Int("serverCount", 3, "number of server nodes")
+var agentCount = flag.Int("agentCount", 1, "number of agent nodes")
+var ci = flag.Bool("ci", false, "running on CI")
+
+// Environment Variables Info:
+// E2E_RELEASE_VERSION=v1.23.1+k3s2 or nil for latest commit from master
+
+func Test_E2ECustomCARotation(t *testing.T) {
+	RegisterFailHandler(Fail)
+	flag.Parse()
+	suiteConfig, reporterConfig := GinkgoConfiguration()
+	RunSpecs(t, "Custom Certificate Rotation Test Suite", suiteConfig, reporterConfig)
+}
+
+var (
+	kubeConfigFile  string
+	agentNodeNames  []string
+	serverNodeNames []string
+)
+
+var _ = ReportAfterEach(e2e.GenReport)
+
+var _ = Describe("Verify Custom CA Rotation", Ordered, func() {
+	Context("Custom CA is rotated:", func() {
+		It("Starts up with no issues", func() {
+			var err error
+			serverNodeNames, agentNodeNames, err = e2e.CreateCluster(*nodeOS, *serverCount, *agentCount)
+			Expect(err).NotTo(HaveOccurred(), e2e.GetVagrantLog(err))
+			fmt.Println("CLUSTER CONFIG")
+			fmt.Println("OS:", *nodeOS)
+			fmt.Println("Server Nodes:", serverNodeNames)
+			fmt.Println("Agent Nodes:", agentNodeNames)
+			kubeConfigFile, err = e2e.GenKubeConfigFile(serverNodeNames[0])
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("Verifies Certificate Rotation", func() {
+			const grepCert = "sudo ls -lt /var/lib/rancher/k3s/server/ | grep tls"
+			var expectResult = []string{"client-ca.crt", "client-ca.key", "client-ca.nochain.crt", "client-ca.pem",
```

Comment:
WDYTA  expectedResult ?  
and formating like this 

`	expectResult := []string{
		"client-ca.crt", "client-ca.key",
		"client-ca.nochain.crt", "client-ca.pem",
		"dynamic-cert.json", "peer-ca.crt",
		"peer-ca.key", "peer-ca.pem",
		"server-ca.crt", "server-ca.key",
		"server-ca.pem", "intermediate-ca.crt",
		"intermediate-ca.key", "intermediate-ca.pem",
		"request-header-ca.crt", "request-header-ca.key",
		"request-header-ca.pem", "root-ca.crt",
		"root-ca.key", "root-ca.pem",
		"server-ca.crt", "server-ca.key",
		"server-ca.nochain.crt", "server-ca.pem",
		"service.current.key", "service.key",
		"apiserver-loopback-client__.crt", "apiserver-loopback-client__.key",
		"",
	}`

Reply:
Done!